config.toml

```diff
@@ -101,6 +101,15 @@ manager_type = "OnDisk"
 
 # Example of an Mbed Crypto provider configuration.
 [[provider]]
+# ⚠
+# ⚠ WARNING: Provider name cannot change.
+# ⚠ WARNING: Choose a suitable naming scheme for your providers now.
+# ⚠ WARNING: Provider name defaults to "mbed-crypto-provider" if not provided, you will not be able to change
+# ⚠ the provider's name from this if you decide to use the default.
+# ⚠
+# (Optional) The name of the provider
+name = "mbed-crypto-provider"
```

Comment:
I am just thinking now if it's really worth it to have optional `name`? In the end most people will copy and adapt the default configuration so they won't lose time thinking about a provider name? And that would save us in some cases where people do not follow the warning notices?

Note that we can go from required to optional but can't do the reverse (for stability).

Reply:
Doesn't our stability guarantee that new config options will be optional? As to ensure that old config files will be valid in new versions of parsec?

Reply:
ah yes you are right, forgot about that point 😁

Reply:
I guess we can add optional sections that have required fields, but not required fields to already existing sections... 😢